scipy/ndimage/filters.py

```diff
@@ -1187,6 +1187,9 @@ def rank_filter(input, rank, size=None, footprint=None, output=None,
     >>> ax2.imshow(result)
     >>> plt.show()
     """
+    if not isinstance(rank, int):
```

Comment:
Might be better to use [`numbers.Integral`]( https://docs.python.org/2/library/numbers.html#numbers.Integral ). That way if someone passed in `numpy.int64` or other integral type, this would still work.

Reply:
done.

Reply:
The standard method we now try to use in SciPy is simply to do:
```
rank = operator.index(rank)
```
It is very general and already gives a sensible error message

Reply:
Neat! Was not aware of that option. Thanks for sharing. :)